crates/ruff_linter/src/rules/ruff/rules/class_with_mixed_type_vars.rs

```diff
@@ -0,0 +1,243 @@
+    fn message(&self) -> String {
+        "Class with type parameter list inherits from `Generic`".to_string()
+    }
+
+    fn fix_title(&self) -> Option<String> {
+        Some("Remove `Generic` base class".to_string())
+    }
+}
+
+/// RUF060
+pub(crate) fn class_with_mixed_type_vars(checker: &mut Checker, class_def: &StmtClassDef) {
+    if checker.settings.target_version < PythonVersion::Py312 {
+        return;
+    }
+
+    let StmtClassDef {
+        type_params,
+        arguments,
+        ..
+    } = class_def;
+
+    let Some(type_params) = type_params else {
+        return;
+    };
+
+    let Some(arguments) = arguments else {
+        return;
+    };
+
+    let Some((generic_base, old_style_type_vars)) =
+        typing_generic_base_and_arguments(arguments, checker.semantic())
+    else {
+        return;
+    };
+
+    let mut diagnostic = Diagnostic::new(ClassWithMixedTypeVars, generic_base.range);
+
+    if let Some(fix) = convert_type_vars(
+        generic_base,
+        old_style_type_vars,
+        type_params,
+        arguments,
+        checker,
+    ) {
+        diagnostic.set_fix(fix);
+    }
+
+    checker.diagnostics.push(diagnostic);
+}
+
+fn typing_generic_base_and_arguments<'a>(
+    class_arguments: &'a Arguments,
+    semantic: &SemanticModel,
+) -> Option<(&'a ExprSubscript, &'a Expr)> {
+    let (_, base @ ExprSubscript { slice, .. }) = find_generic(class_arguments, semantic)?;
+
+    Some((base, slice.as_ref()))
+}
+
+fn convert_type_vars(
+    generic_base: &ExprSubscript,
+    old_style_type_vars: &Expr,
+    type_params: &TypeParams,
+    class_arguments: &Arguments,
+    checker: &Checker,
+) -> Option<Fix> {
+    let mut type_vars = type_params
+        .type_params
+        .iter()
+        .map(TypeVar::from)
+        .collect::<Vec<_>>();
+
+    let semantic = checker.semantic();
+    let mut converted_type_vars = match old_style_type_vars {
+        Expr::Tuple(ExprTuple { elts, .. }) => {
+            generic_arguments_to_type_vars(elts.iter(), type_params, semantic)?
+        }
+        expr @ (Expr::Subscript(_) | Expr::Name(_)) => {
+            generic_arguments_to_type_vars(iter::once(expr), type_params, semantic)?
+        }
+        _ => return None,
+    };
+
+    type_vars.append(&mut converted_type_vars);
+
+    let source = checker.source();
+    let new_type_params = DisplayTypeVars {
+        type_vars: &type_vars,
+        source,
+    };
+
+    let remove_generic_base =
+        remove_argument(generic_base, class_arguments, Parentheses::Remove, source).ok()?;
+    let replace_type_params =
+        Edit::range_replacement(new_type_params.to_string(), type_params.range);
+
+    Some(Fix::unsafe_edits(
+        remove_generic_base,
+        [replace_type_params],
+    ))
+}
+
+fn generic_arguments_to_type_vars<'a>(
+    exprs: impl Iterator<Item = &'a Expr>,
+    existing_type_params: &TypeParams,
+    semantic: &'a SemanticModel,
+) -> Option<Vec<TypeVar<'a>>> {
```

Comment:
Could you add some documentation to this function to explain why it sometimes returns `None`? I always do a double take when I see a function that returns `Option<Vec<_>>`, because I always wonder why it doesn't just return an empty `Vec` instead of returning `None`. What does returning `None` signify here that wouldn't be expressed by returning an empty `Vec`?